### PR TITLE
Allow Passing Tags in E/T/TC Create methods

### PR DESCRIPTION
**Pull request by yzhu0**
*Description of changes:*
- Allow Passing Tags in E/T/TC Create methods

*Issue #, if available:*
- https://sim.amazon.com/issues/AML-71615

*Testing done:*
- Unit tests & integration tests

### CODE REVIEW

File: src/smexperiments/experiment.py [hunk 1]
```diff
@@ -101,6 +103,7 @@ def create(cls, experiment_name=None, description=None, sagemaker_boto_client=No
             experiment_description: (str, optional): Description of the experiment
             sagemaker_boto_client (SageMaker.Client, optional): Boto3 client for SageMaker. If not
                 supplied, a default boto3 client will be created and used.
+            tags (List[dict[str, str]]): A list of tags to associate with the experiment.
```
Comment (danabens) [inline src/smexperiments/experiment.py:106]: should just be ``` tags (dict): A dictionary of ....``` List of dict implies there are multiple dictionaries.

File: src/smexperiments/trial.py [hunk 2]
```diff
@@ -43,10 +43,12 @@ class Trial(_base_types.Record):
     Attributes:
         trial_name (str): The name of the trial.
         experiment_name (str): The name of the trial's experiment.
+        tags (List[dict[str, str]]): A list of tags to associate with the trial.
```
Comment (danabens) [inline src/smexperiments/trial.py:46]: same, just dict

File: src/smexperiments/trial.py [hunk 3]
```diff
@@ -96,15 +98,16 @@ def load(cls, trial_name, sagemaker_boto_client=None):
...
         """Create a new trial and return a ``Trial`` object.
 
         Args:
             experiment_name: (str): Name of the experiment to create this trial in.
             trial_name: (str, optional): Name of the Trial. If not specified, an auto-generated name will be used.
             sagemaker_boto_client (SageMaker.Client, optional): Boto3 client for SageMaker.
                 If not supplied, a default boto3 client will be created and used.
-            trial_components (list): A list of trial component names, trial components, or trial component trackers
+            trial_components (list): A list of trial component names, trial components, or trial component trackers.
+            tags (List[dict[str, str]]): A list of tags to associate with the trial.
```
Comment (danabens) [inline src/smexperiments/trial.py:110]: just dict

File: tests/integ/test_trial_component.py [hunk 4]
```diff
@@ -23,6 +23,10 @@ def test_create_delete(trial_component_obj):
     assert trial_component_obj.trial_component_name
 
 
+def test_create_tags(trial_component_obj):
+    assert trial_component_obj.tags
```
Comment (danabens) [inline tests/integ/test_trial_component.py:27]: this looks like this is just testing that the test TC has the tags populated. instead add tags to the existing test_save integ test here. same thing in trial and experiment integ tests.

File: src/smexperiments/experiment.py (hunk 1 above)
Comment (yzhu0) [reply src/smexperiments/experiment.py:106]: Based on the tag under create_trial_component https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html?highlight=experiment#SageMaker.Client.create_trial_component, should it be a list of dict?

Comment (danabens) [reply src/smexperiments/experiment.py:106]: looks like a list of dict is the normal way to represent tags https://github.com/aws/sagemaker-python-sdk/search?p=3&q=tags&unscoped_q=tags so go with it

File: tests/integ/test_trial_component.py (hunk 4 above)
Comment (yzhu0) [reply tests/integ/test_trial_component.py]: updated the integ test with actual tag check (calling sagemaker_boto_client.list_tags() check)

**Pull request merged**